### PR TITLE
Feature: add custom network

### PR DESCRIPTION
**Pull request by bogos**
- [x]  Add custom network modal selector
- [ ]  Validate selected network

### CODE REVIEW

**Comment by 0xLucca** [comment]
The code in the previous commit is functional but appears messy and lacks clarity due to asynchronous operations within useEffect. Despite its smelly nature, it successfully initializes the API and manages state. Consider refactoring for improved readability when possible.

File: src/hooks/useNetworkApi.ts [hunk 1]
```diff
@@ -1,24 +1,53 @@
...
   const { networkConnected } = useNetworkAccountsContext()
-  const api = useApi(networkConnected)
-  const firstLoadCompleted = useDelay(5000)
+  const [api, setApi] = useState<ApiPromise | undefined>(undefined)
 
+  const fetchApi = useApi(networkConnected)
+
+  useEffect(() => {
+    if (networkConnected) {
+      if (!fetchApi) {
```
Comment (henrypalacios) [inline src/hooks/useNetworkApi.ts:37]: 🤔 This logic is not stable, what happen if `fetchApi` returns after `useEffect` is executed?

**Review by henrypalacios — APPROVED** [review]
Great changes @bogos 🤝.

I will merge and leave you the commentary on what needs to be improved.

File: src/view/components/WalletConnectButton/NetworkSelect.tsx [hunk 2]
```diff
@@ -60,39 +77,98 @@ const StyledMenuItem = styled(MenuItem)<MenuItemProps>(() => ({
...
-      onChange(chainId)
+      onChange(chainId as ChainId)
     } else {
-      openModal()
+      openDialog()
     }
   }
 
+  const formData = {
+    name: useFormInput<string>('test', [notEmpty]),
```
Comment (0xLucca) [inline src/view/components/WalletConnectButton/NetworkSelect.tsx:125]: Here in the form data the placeholder for the name seems like an address

**Review by 0xLucca — APPROVED** [review]
The logic works as expected. There are some minor issues to fix in the UI

**Pull request merged**